substrate/frame/support/src/storage/mod.rs

```diff
@@ -167,6 +167,27 @@ pub trait StorageValue<T: FullCodec> {
 	{
 		T::decode_len(&Self::hashed_key())
 	}
+
+	/// Read the length of the storage value without decoding the entire value.
+	///
+	/// `T` is required to implement [`StorageDecodeNonDedupLength`].
+	///
+	/// If the value does not exists or it fails to decode the length, `None` is returned.
+	/// Otherwise `Some(len)` is returned.
+	///
+	/// # Warning
+	///
+	/// - The value returned is the non-deduplicated length of the underlying Vector in storage.
+	///
+	/// - `None` does not mean that `get()` does not return a value. The default value is completely
+	/// ignored by this function.
+
+	fn decode_dedup_len() -> Option<usize>
```

Comment:
would suggest an example, in rust-docs, demonstrating under which conditions this returns a value different than `decode_len`

Reply:
Good idea I will add this.

Reply:
actually reading the PR description again, it will always return the same value right? it is only a new marker trait. Nonetheless some examples would be good. Also see https://github.com/paritytech/substrate/blob/master/docs/DOCUMENTATION_GUIDELINES.md

Reply:
Just re-read your comment. I renamed the method to the `decode_non_dedup_len`.

They both, `decode_len` and `decode_non_dedup_len`, always return the same value, the new trait simply improves semantics as per: https://github.com/paritytech/polkadot-sdk/issues/126#issuecomment-1691677905.